CMakeLists.txt

```diff
@@ -125,195 +129,130 @@ else()
     set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 endif()
 
-set(AVIF_PLATFORM_DEFINITIONS)
-set(AVIF_PLATFORM_INCLUDES)
-set(AVIF_PLATFORM_SYSTEM_INCLUDES)
-set(AVIF_PLATFORM_LIBRARIES)
+add_library(avif)
 
-# A macro that wraps find_package, preferring static libraries if this is a static build
-# Should only be used for libraries linked against avif, not for those only used
-# by avif_apps or tests
-macro(find_package_libavif)
-    set(_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES, "${CMAKE_FIND_LIBRARY_SUFFIXES}")
-    if(NOT BUILD_SHARED_LIBS)
-        if(WIN32 OR MINGW)
-            set(CMAKE_FIND_LIBRARY_SUFFIXES .a .lib ${CMAKE_FIND_LIBRARY_SUFFIXES})
-        else()
-            set(CMAKE_FIND_LIBRARY_SUFFIXES .a ${CMAKE_FIND_LIBRARY_SUFFIXES})
+function(avif_target_link_library target)
```

Comment:
Please comment what this functions does.

Reply:
Done.